plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableMetaQuery.php

```diff
@@ -0,0 +1,566 @@
+<?php
+namespace Automattic\WooCommerce\Internal\DataStores\Orders;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class used to implement meta queries for the orders table datastore via {@see OrdersTableQuery}.
+ * Heavily inspired by WordPress' own `WP_Meta_Query` for backwards compatibility reasons.
+ *
+ * Parts of the implementation have been adapted from {@link https://core.trac.wordpress.org/browser/tags/6.0.1/src/wp-includes/class-wp-meta-query.php}.
+ */
+class OrdersTableMetaQuery {
```

Comment:
nitpick: bunch of functions such as `get_sql_clauses`, `is_atomic` etc don't have their return type declared, but can be declared safely.

Reply:
Done.